src/Microsoft.TemplateEngine.Abstractions/ITemplateEngineHost.cs

```diff
@@ -18,8 +18,6 @@ public interface ITemplateEngineHost
 
         void LogTiming(string label, TimeSpan duration, int depth);
 
-        string Locale { get; }
```

Comment:
just a thought: should we rather keep the property but rather force it to be CultureInfo.CurrentUICulture.Name and UpdateLocale to update UICulture when changed in our implementations of hosts?

random idea: some author host might want to implement Locale differently from CultureInfo.CurrentUICulture and this way they will still be able to do it. 
Also I'm not fan of Edge calling CultureInfo for file naming determining.

Reply:
Changing the `CurrentUICulture` inside `UpdateLocale` might have undesired effects. The caller will think that it is changing the locale of the host, but it is changing it for the whole thread. Someone may create 2 different hosts and give them different locales, which again won't work.

For your random idea: Localized resources in .resx will still use CurrentUICulture by default. If someone overrides the default locale, resource managers will need to be updated as well. For me, it sounds simple to have everything get locale from CurrentUICulture.

What part of the code are you talking about with "Edge calling CultureInfo"? I'm open to suggestions.